src/lv_draw/lv_img_cache.c

```diff
@@ -33,13 +33,15 @@
 /**********************
  *  STATIC PROTOTYPES
  **********************/
-#if LV_IMG_CACHE_DEF_SIZE == 0
-    static lv_img_cache_entry_t cache_temp;
-#endif
+static bool lv_img_cache_match(const void * src1, const void * src2);
 
 /**********************
  *  STATIC VARIABLES
  **********************/
+#if LV_IMG_CACHE_DEF_SIZE == 0
+    static lv_img_cache_entry_t cache_temp;
```

Comment:
Shouldn't this be a GC root?

Reply:
As memory can be allocated inside `cache_temp` it really should be GC root. (Similarly to [_lv_img_cache_array](https://github.com/lvgl/lvgl/blob/351008edcc01753018ecc6a7a309fa01f2d42a81/src/lv_misc/lv_gc.h#L38))

But we can't use `#if` inside define so disabled cache config should be rather handled via a `_lv_img_cache_array[1]` array. I'll add this to the PR in a few minutes.

Reply:
@kisvegabor I am investigating the solution too.

Reply:
Added in eb8bfb9

Reply:
@xiaoxiang781216 Let me know if you have  a better idea

Reply:
@kisvegabor I incoperate your change with minor modification, please take a look.